samples/tacos/src/main/kotlin/com/slack/circuit/tacos/OrderTacosCircuit.kt

```diff
@@ -0,0 +1,356 @@
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.graphics.vector.rememberVectorPainter
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.slack.circuit.runtime.CircuitUiEvent
+import com.slack.circuit.runtime.CircuitUiState
+import com.slack.circuit.runtime.Screen
+import com.slack.circuit.runtime.presenter.Presenter
+import com.slack.circuit.tacos.model.Diet
+import com.slack.circuit.tacos.model.Ingredient
+import com.slack.circuit.tacos.step.ConfirmationOrderStep
+import com.slack.circuit.tacos.step.ConfirmationUi
+import com.slack.circuit.tacos.step.FillingsOrderStep
+import com.slack.circuit.tacos.step.FillingsUi
+import com.slack.circuit.tacos.step.OrderStep
+import com.slack.circuit.tacos.step.SummaryOrderStep
+import com.slack.circuit.tacos.step.SummaryUi
+import com.slack.circuit.tacos.step.ToppingsOrderStep
+import com.slack.circuit.tacos.step.ToppingsUi
+import com.slack.circuit.tacos.ui.theme.TacoTheme
+import java.math.BigDecimal
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.persistentSetOf
+import kotlinx.collections.immutable.toImmutableSet
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+object OrderTacosScreen : Screen {
+  data class State(
+    @StringRes val headerResId: Int,
+    val orderCost: BigDecimal,
+    val stepState: OrderStep.State,
+    val isPreviousVisible: Boolean,
+    val isNextEnabled: Boolean,
+    val isNextVisible: Boolean,
+    val isFooterVisible: Boolean,
+    val eventSink: (Event) -> Unit
+  ) : CircuitUiState
+
+  sealed interface Event : CircuitUiEvent {
+    val indexModifier: Int
+      get() = 0
+
+    object Previous : Event {
+      override val indexModifier: Int = -1
+    }
+    object Next : Event {
+      override val indexModifier: Int = 1
+    }
+    object ProcessOrder : Event {
+      override val indexModifier: Int = 1
+    }
+  }
+}
+
+@Immutable
+data class OrderDetails(
+  val filling: Ingredient = Ingredient(""),
+  val toppings: Set<Ingredient> = persistentSetOf(),
+  val baseCost: BigDecimal = BigDecimal("9.99"), // Default taco price
+  val ingredientsCost: BigDecimal = BigDecimal.ZERO
```

Comment:
`BigDecimal`/floating point in general is an anti-pattern for currency. Let's just use an int that represents cents, then make the render string format it

Reply:
Yeah...I started out using cents (ie. an `Int`) then switched to `BigDecimal` to simplify conversion to dollars in the UI.

Reintroduced `Cents` typealias and added a extension to handle conversion to dollars. https://github.com/slackhq/circuit/pull/496/commits/53e915711ad82900bcfe50764d6511b5f1987d26